File: SQLSchemaCompare.UI/wwwroot/js/Entities/ApiResponse.ts
/**
 * Represent an Api response
 */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
class ApiResponse<T> {

    /**
     * Indicates whether a response is success
     */
    public Success?: boolean;

    /**
     * The error code
     */
    public ErrorCode?: ApiErrorCode;

    /**
     * The error message
     */
    public ErrorMessage?: string;

    /**
     * The error message
     */
    public Result?: T;

}

File: SQLSchemaCompare.Core/Entities/Api/ApiResponse.cs
namespace TiCodeX.SQLSchemaCompare.Core.Entities.Api
{
    /// <summary>
    /// Represents the response for Api requests
    /// </summary>
    public class ApiResponse
    {
        /// <summary>
        /// Gets or sets a value indicating whether the request succeeded or returned an error
        /// </summary>
        public bool Success { get; set; } = true;

        /// <summary>
        /// Gets or sets the response error code
        /// </summary>
        public EErrorCode ErrorCode { get; set; } = EErrorCode.Success;

        /// <summary>
        /// Gets or sets the response error message
        /// </summary>
        public string ErrorMessage { get; set; } = string.Empty;
    }
}
